discovery-provider/src/tasks/index_aggregate_monthly_plays.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import time
+
+from sqlalchemy import func, text
+from src.models import Play
+from src.tasks.celery_app import celery
+from src.utils.update_indexing_checkpoints import (
+    get_last_indexed_checkpoint,
+    save_indexed_checkpoint,
+)
+
+logger = logging.getLogger(__name__)
+
+AGGREGATE_MONTHLY_PLAYS_TABLE_NAME = "aggregate_monthly_plays"
+
+# UPSERT_AGGREGATE_MONTHLY_PLAYS_QUERY
+# Get new plays that came after the last indexing checkpoint for aggregate_monthly_play
+# Group those new plays by play item id and month to get the aggregate monthly play counts
+# For new play item ids, insert those aggregate counts
+# For existing play item ids, add the new aggregate count to the existing aggregate count
+UPSERT_AGGREGATE_MONTHLY_PLAYS_QUERY = """
+    with aggregate_monthly_plays_last_checkpoint as (
+        select
+            :prev_id_checkpoint as prev_id_checkpoint,
+            :new_id_checkpoint as new_id_checkpoint
+    ),
+    new_plays as (
+        select
+            play_item_id,
+            date_trunc('month', created_at) as timestamp,
+            count(play_item_id) as count
+        from
+            plays p
+        where
+            p.id > (
+                select
```

Comment:
do you need these subqueries since this is being tracked in indexing checkpoints and passed in?

Reply:
Oh yeah it's not really necessary. In fact the upper bound is not really needed either.